docs/specs/editorconfig-in-the-compiler.md

```diff
@@ -0,0 +1,110 @@
+All-capital words are interpreted per [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+# Compiler Command Arguments
+
+The compiler *MUST* receive a new switch, `/editorconfig`, which accepts a single
+file which is an .editorconfig file that may apply to one or more files in the
+compilation. Multiple files are passed with multiple `/editorconfig` switches,
+the same way you can pass multiple references with /reference or multiple analyzers
+with /analyzer. There is no short form of the argument. Files passed with relative
+file names are resolved relative to the current working directory the same
+as source files.
+
+Examples:
+
+```
+    csc.exe /editorconfig:.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:..\.editorconfig
+    csc.exe /editorconfig:.editorconfig /editorconfig:Subdirectory\.editorconfig
+```
+
+It is expected that the compiler is given the .editorconfig files with casing that
+matches the underlying casing of the file system. The compiler *MUST NOT* do any case correction. So
+
+```
+    csc.exe /editorconfig:Subdirectory\.editorconfig /editorconfig:SubDirectory\.editorconfig
+```
+
+is a valid input to the compiler. The compiler *MAY* choose to issue a warning if two paths given
+are exactly equal by ordinal in a case-sensitive way.
+
+If multiple .editorconfigs apply to a given file (the precise definition of
```

Comment:
@jaredpar If the only way the compiler was launched is through code we control, then it would be a neutral change either way. This spec is slightly cleaner if the arguments are unordered, while the compiler implementation is slightly cleaner if the arguments are ordered.

However, the compiler is a command line application that can be invoked in other ways. Considering there is a very specific expectation for behavior of EditorConfig with respect to the paths of its input arguments, the experience for end users is substantially improved by placing the ordering requirement on the compiler and not the command line. This is an easy win for end users.